tests/callbacks/test_model_checkpoint.py

```diff
@@ -104,26 +104,32 @@ def test_model_checkpoint_save_last_checkpoint_contents(tmpdir):
     num_epochs = 3
     model_checkpoint = ModelCheckpoint(filepath=tmpdir, save_top_k=num_epochs, save_last=True)
     trainer = Trainer(
-        default_root_dir=tmpdir,
-        early_stop_callback=False,
-        checkpoint_callback=model_checkpoint,
-        max_epochs=num_epochs,
+        default_root_dir=tmpdir, early_stop_callback=False, checkpoint_callback=model_checkpoint, max_epochs=num_epochs,
     )
     trainer.fit(model)
     path_last_epoch = model_checkpoint.format_checkpoint_name(num_epochs - 1, {})  # epoch=3.ckpt
     path_last = str(tmpdir / ModelCheckpoint.CHECKPOINT_NAME_LAST)  # last.ckpt
     assert path_last_epoch != path_last
     ckpt_last_epoch = torch.load(path_last_epoch)
     ckpt_last = torch.load(path_last)
-    matching_keys = (
+
+    trainer_keys = (
         "epoch",
         "global_step",
-        ModelCheckpoint.CHECKPOINT_STATE_BEST_SCORE,
-        ModelCheckpoint.CHECKPOINT_STATE_BEST_PATH,
     )
-    for key in matching_keys:
+    for key in trainer_keys:
         assert ckpt_last_epoch[key] == ckpt_last[key]
 
+    checkpoint_callback_keys = (
+        "best_model_score",
+        "best_model_path",
+    )
+    for key in checkpoint_callback_keys:
+        assert (
+            ckpt_last_epoch["callbacks"][type(model_checkpoint)][key]
```

Comment:
@jeremyjordan this check will always be true. Was this a copy/paste issue?  Did you mean `ckpt_last` here?